stdlib/universe/fill.go

```diff
@@ -167,32 +172,37 @@ func createFillTransformation(id execute.DatasetID, mode execute.AccumulationMod
 	if !ok {
 		return nil, nil, errors.Newf(codes.Internal, "invalid spec type %T", spec)
 	}
-	cache := execute.NewTableBuilderCache(a.Allocator())
-	d := execute.NewDataset(id, mode, cache)
-	t := NewFillTransformation(d, cache, s)
+	t, d := NewFillTransformation(a.Context(), s, id, a.Allocator())
 	return t, d, nil
 }
 
 type fillTransformation struct {
-	d     execute.Dataset
-	cache execute.TableBuilderCache
-
-	spec *FillProcedureSpec
+	d     *execute.PassthroughDataset
+	ctx   context.Context
+	spec  *FillProcedureSpec
+	alloc *memory.Allocator
 }
 
-func NewFillTransformation(d execute.Dataset, cache execute.TableBuilderCache, spec *FillProcedureSpec) *fillTransformation {
-	return &fillTransformation{
-		d:     d,
-		cache: cache,
+func NewFillTransformation(ctx context.Context, spec *FillProcedureSpec, id execute.DatasetID, alloc *memory.Allocator) (*fillTransformation, execute.Dataset) {
```

Comment:
Can you update this signature so it returns `(execute.Transformation, execute.Dataset)`? I prefer not to have public methods that expose private structs.